client/src/components/Tea/TeaCollectionTable/TeaCollectionTableContainer.jsx

```diff
@@ -0,0 +1,218 @@
+import React from "react";
+import { connect } from "react-redux";
+import { deleteTea, getTeas } from "../../../actions/teaActions";
+import TeaCollectionTable from "./TeaCollectionTable";
+import DataList from "../../DataList";
+
+export class TeaCollectionTableContainer extends React.Component {
+  state = {
+    sortedIDs: [],
+    memoizedIDs: {},
+    sortColumn: "",
+    sortOrder: "",
+    formControls: {
+      filterCategory: "",
+      filterCriteria: ""
+    },
+    filtered: false,
+    dataList: []
+  };
+
+  columnHeaders = [
+    { colName: "name", colTitle: "Name" },
+    { colName: "brand", colTitle: "Brand" },
+    { colName: "teaType", colTitle: "Type" },
+    { colName: "servings", colTitle: "Servings" }
+  ];
+
+  handleDeleteClick = tea => this.props.handleDelete(tea);
+
+  handleSortClick = (columnName, sortOrder) => {
+    let newState = {
+      sortColumn: columnName,
+      sortOrder: sortOrder
+    };
+
+    if (this.state.memoizedIDs.columnName === columnName) {
+      newState["sortedIDs"] = this.state.memoizedIDs[sortOrder];
+    } else {
+      let newSortOrder = this.state.sortedIDs.sort((a, b) => {
+        if (
+          this.props.teas.allTeas[a][columnName] >
+          this.props.teas.allTeas[b][columnName]
+        ) {
+          return 1;
+        }
+        if (
+          this.props.teas.allTeas[a][columnName] <
+          this.props.teas.allTeas[b][columnName]
+        ) {
+          return -1;
+        }
+        return 0;
+      });
+
+      const revSortOrder = [...newSortOrder].reverse();
+
+      newState["sortedIDs"] = newSortOrder;
+
+      newState["memoizedIDs"] = {
+        columnName: columnName,
+        asc: newSortOrder,
+        desc: revSortOrder
+      };
+    }
+    this.setState({
+      ...newState
+    });
+  };
+
+  sortColumnHandler = columnName =>
+    this.state.sortColumn === columnName && this.state.sortOrder === "asc";
+
+  filterDropdownChangeHandler = event => {
+    const dataListOptions = this.state.sortedIDs.map(
```

Comment:
You're still getting an error when you filter by something, then return the filter dropdown to the "Filter By" option.  You can handle this here by checking for when  `event.target.value === ''`, then calling `handleClearFilterClick` if that's the case.  Also, not a bad idea to turn `event.target.value` into a variable, since it's used in a couple places.